src/utils/object.ts

```diff
@@ -5,35 +5,33 @@
 
 export type Transformer = (...args: any[]) => string;
 
-export function transformKeys(obj: Record<string, any>, transformer: Transformer, ignoreKeys: string[] = []): any {
+export const transformKeys = (
+    obj: Record<string, any> | unknown[],
+    transformer: Transformer,
+    ignoreKeys: string[] = []
+): any => {
     const isObject = (value: unknown): value is Record<string, any> => typeof value === "object" && Boolean(value);
 
-    const transformArray = <T>(arr: T[]): T[] =>
+    const transformArray = (arr: unknown[], transformer: Transformer): any =>
         arr.map((item) => (isObject(item) ? transformKeys(item, transformer) : item));
 
     if (Array.isArray(obj)) {
-        return transformArray(obj);
+        return transformArray(obj, transformer);
     }
 
+    const ignoredKeySet = new Set(ignoreKeys || []);
     return Object.keys(obj || {}).reduce((acc: Record<string, any>, key: string) => {
-        let value = obj[key];
+        const value = obj[key];
 
-        if (ignoreKeys.includes(key)) {
+        if (ignoredKeySet.has(key)) {
             return { ...acc, [key]: value };
         }
 
-        if (isObject(value)) {
-            if (Array.isArray(value)) {
-                value = transformArray(value);
-            } else {
-                value = transformKeys(value, transformer);
-            }
-        }
-
-        acc[transformer(key)] = value;
```

Comment:
just wondering, the logic changed here a little, is it an improvement or bugfix? 🤔

Reply:
The function already handles the arrays 🙂 Which is why the logic is still the same. The array handling was repeated.

Also made the code immutable in this function (except for `acc` as per convention)

Reply:
Changed the `obj[key]` back to `value` 🙂 Thanks!

Reply:
The linked test to my first point is the array test:

```
myKey2: [
                {
                    mySubKey21: "Value 4-1",
                    mySubKey22: "Value 4-2",
                    mySubKey23: ["Value 4-3-1", "Value 4-3-2"],
                },
            ],
```